v0/destinations/blueshift/transform.js

```diff
@@ -0,0 +1,166 @@
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  constructPayload,
+  ErrorMessage,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getValueFromMessage,
+  getFieldValueFromMessage,
+  removeUndefinedAndNullValues,
+  defaultPutRequestConfig
+} = require("../../util");
+
+const {
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES,
+  BASE_URL_EU,
+  BASE_URL
+} = require("./config");
+
+function checkValidEventName(str) {
+  if (
+    str.indexOf(".") !== -1 ||
+    str.indexOf(" ") !== -1 ||
+    /[0-9]/.test(str) ||
+    str.length > 64
+  )
+    return true;
+  return false;
+}
+
+const trackResponseBuilder = async (message, category, { Config }) => {
+  let event = getValueFromMessage(message, "event");
+  if (!event) {
+    throw new CustomError(
+      "[Blueshift] property:: event is required for track call",
+      400
+    );
+  }
+  event = event.trim().toLowerCase();
+  if (checkValidEventName(event)) {
+    throw new CustomError(
+      "[Blueshift] Event name doesn't contain period(.), whitespace, numeric value and contains not more than 64 characters",
+      400
+    );
+  }
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
```

Comment:
Blueshift Supports ecom spec Ref: https://developer.blueshift.com/docs/your-sites-event-data#add-to-cart
Rudderstack also supports ecom spec Ref: https://www.rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-ecommerce-events-specification/ordering/

We should also support specific mapping for these